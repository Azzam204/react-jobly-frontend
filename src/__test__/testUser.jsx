import React from "react";
import UserContext from "../userContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
  ({ children, currUser = demoUser, isApplied = () => false }) => (
    <UserContext.Provider value={{ currUser, isApplied }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider }