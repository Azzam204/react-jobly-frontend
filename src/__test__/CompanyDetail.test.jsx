import React from "react";
import { render } from "@testing-library/react";
import { UserProvider } from "./testUser";
import { MemoryRouter } from "react-router-dom";
import Company from "../CompanyDetail";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Company />
      </UserProvider>
    </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Company />
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});


