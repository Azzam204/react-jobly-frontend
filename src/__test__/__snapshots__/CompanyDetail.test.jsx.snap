// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-8 offset-md-2"
  >
    <h4
      class="white"
    />
    <p
      class="white"
    />
    <div />
  </div>
</DocumentFragment>
`;
